partner_compassion/models/partner_compassion.py

```diff
@@ -23,6 +23,32 @@ class ResPartner(models.Model):
     """
     _inherit = 'res.partner'
 
+    @api.multi
+    @api.depends('category_id')
+    def _compute_has_sponsorships(self):
+        """
+        A partner is sponsor if he is correspondent of at least one
+        sponsorship.
+        """
+        for partner in self:
+            if partner.is_church:
+                partner.has_sponsorships = self.env[
+                    'recurring.contract'].search_count([
+                        '|',
+                        ('partner_id', 'in', partner.member_ids.ids),
+                        ('correspondant_id', 'in', partner.member_ids.ids),
+                        ('type', 'like', 'S')
+                    ])
+                partner.number_sponsorships = self.env[
```

Comment:
I think you can use the same search for number_sponsorships and has_sponsorships. Just do
```
nb_sponsorships = search_count([...])
partner.nb_sponsorships = nb_sponsorships
partner.has_sponsorships = nb_sponsorships
```
You can as well correct it in compassion-modules where I believe it's as well duplicated for nothing